yt/visualization/_handlers.py

```diff
@@ -327,45 +327,19 @@ def get_norm(self, data: np.ndarray, *args, **kw) -> Normalize:
             # allowing to toggle between lin and log scaling without detailed user input
             norm_type = self.norm_type
         else:
-            if kw["vmin"] == kw["vmax"] or not np.any(np.isfinite(data)):
+            if kw["vmin"] == kw["vmax"] or not np.any(finite_values_mask):
                 norm_type = Normalize
             elif kw["vmin"] <= 0:
                 norm_type = SymLogNorm
-            elif (
-                Version("3.3") <= MPL_VERSION < Version("3.5")
```

Comment:
Two reasons: first is that the preceeding `kw["vmin"] <= 0"` condition would always catch before this one (if `kw['vmin']==0` then the preceeding `kw['vmin'] <= 0` will always be True) . I considered flagging it in your PR but since it's closely related to symlog/linthresh settings it seemed better to address it here. 

The second reason I went for removal instead of changing the condition is that controlling for the issue when the range is too large happens down in the linthresh calculation so this seemed redundant. Thinking about it now though, I do think it would actually make sense to allow `LogScale`  when possible... but I think this condition probably needs some adjusting rather than straight removal.

I also mean to test this more thoroughly will different matplotlib versions before switching from draft -- so far just been using the latest release (3.5.2).

Reply:
okay sounds good !

Reply:
So I'm back to thinking I'd rather remove these lines... they're not needed in that sense that removing them does not result in errors. But it is true we'd be switching to SymLogNorm in situations that could technically use a LogNorm -- If we wanted to, we could allow a LogNorm when the data is >= 0 if matplotlib >= 3.5.2 **and** if we supply that new interpolation_stage keyword to imshow. So happy to implement that here if you think it's a good idea, but i'd rather wait til your #3849 is merged before doing so.